### PR TITLE
chore(trading): fees e2e tests

### LINKED ISSUE #5451: E2E tests for fees
Create market-sim E2E tests for fees page, fees on deal ticket. To cover volume discount program and referrral program.

### PR DESCRIPTION
**Pull request by bwallacee**
closes #5451

### CODE REVIEW

File: apps/trading/components/card/card.tsx [hunk 1]
```diff
@@ -86,11 +89,16 @@ export const CardTable = (props: HTMLProps<HTMLTableElement>) => {
   );
 };
 
-export const CardTableTH = (props: HTMLProps<HTMLTableHeaderCellElement>) => {
+interface CardTableTHProps extends HTMLProps<HTMLTableHeaderCellElement> {
```
Comment (mattrussell36) [inline apps/trading/components/card/card.tsx:92]: I don't think you need to extends the props with testId: string. data-testid is a valid normal HTMLProp so the component should accept just passing data-testid as a prop directly

File: apps/trading/components/fees-container/table.tsx [hunk 2]
```diff
@@ -3,21 +3,49 @@ import type { ReactNode } from 'react';
 
 const cellClass = 'px-4 py-2 text-xs font-normal text-left last:text-right';
 
-export const Th = ({ children }: { children?: ReactNode }) => {
+export const Th = ({
+  children,
+  testId,
+}: {
+  children?: ReactNode;
+  testId?: string;
```
Comment (mattrussell36) [inline apps/trading/components/fees-container/table.tsx:11]: I would just type these props as `HTMLProps<HTMLTableHeaderCellElement>` and pass data-testid directly. Same for the other table element components

File: apps/trading/components/fees-container/fees-container.tsx [hunk 3]
```diff
@@ -461,29 +483,37 @@ const VolumeTiers = ({
...
             const isUserTier = tierIndex === i;
 
             return (
-              <Tr key={i}>
-                <Td>{i + 1}</Td>
-                <Td>{formatPercentage(Number(tier.volumeDiscountFactor))}%</Td>
-                <Td>{formatNumber(tier.minimumRunningNotionalTakerVolume)}</Td>
-                <Td>{isUserTier ? formatNumber(lastEpochVolume) : ''}</Td>
-                <Td>{isUserTier ? <YourTier /> : null}</Td>
+              <Tr key={i} testId={`tier-row-${i}`}>
```
Comment (mattrussell36) [inline apps/trading/components/fees-container/fees-container.tsx:503]: (non blocking): I find it a tad overkill here to add a testid for everything in the table, including the index. Seeing as you'll presumably be using the index in the test code anyway would it be cleaner to just grab all the rows and reference them by index from the resulting array?

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/components/card/card.tsx (hunk 1 above)
Comment (bwallacee) [reply apps/trading/components/card/card.tsx]: Fixed

File: apps/trading/components/fees-container/table.tsx (hunk 2 above)
Comment (bwallacee) [reply apps/trading/components/fees-container/table.tsx]: Fixed

File: apps/trading/e2e/tests/market/test_market_selector.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 @pytest.mark.usefixtures("page", "continuous_market", "auth", "risk_accepted")
 def test_market_selector(continuous_market, page: Page):
     page.goto(f"/#/markets/{continuous_market}")
+    page.pause()
```
Comment (dalebennett1992) [inline apps/trading/e2e/tests/market/test_market_selector.py:8]: is this rogue

Comment (bwallacee) [reply apps/trading/e2e/tests/market/test_market_selector.py]: Removed

File: apps/trading/e2e/tests/fees/test_fees.py [hunk 5]
```diff
@@ -0,0 +1,473 @@
...
+    else:
+        expect(page.get_by_test_id(REQUIRED_EPOCHS_VALUE_0)).to_have_text("1")
+        expect(page.get_by_test_id(REQUIRED_EPOCHS_VALUE_1)).to_have_text("2")  
+
+    expect(page.get_by_test_id(your_tier)).to_be_visible()
+    expect(page.get_by_test_id(your_tier)).to_have_text("Your tier")
+
+@pytest.mark.parametrize("tier, discount_program, fees_after_discount", [
+    (1, "volume", "9.045%"),
+    #(2, "volume", "8.04%"),
```
[the remark this answers is not in the document]
Comment (bwallacee) [reply apps/trading/e2e/tests/fees/test_fees.py]: Removed

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading/e2e/tests/fees/test_fees.py [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+
+    order_count = 2
+    order_size = 2
+    for _ in range(order_count):
+        submit_order(vega, "Key 1", market, "SIDE_BUY", order_size, 110)
+        forward_time(vega, True if _ < order_count - 1 else False)
+    return market
+
+def set_market_volume_discount(vega, tier, discount_program):
+    global market_id_tier_1_volume
```
[the remark this answers is not in the document]
Comment (bwallacee) [reply apps/trading/e2e/tests/fees/test_fees.py]: Removed.

**Review by mattrussell36 — APPROVED** [review]

**Review by daro-maj — APPROVED** [review]

**Pull request merged**